Marvelo/Libs/DataStoreLive/Sources/DataStoreLive.swift

```diff
@@ -149,5 +169,36 @@ extension DataStoreLive: DataStore {
 				self.events = events
 			} catch {}
 	}
-	
+
+    // MARK: - AsyncStreams for subscribing to data changes
+
+    public func charactersStream() -> AsyncStream<[Character]> {
+        let continuationID = UUID()
+        return AsyncStream { continuation in
+            charactersStreamContinuations[continuationID] = continuation
+            continuation.onTermination = { [weak self] _ in
+                self?.charactersStreamContinuations[continuationID] = nil
+            }
+        }
+    }
+
+    public func eventsStream() -> AsyncStream<[Event]> {
+        let continuationID = UUID()
+        return AsyncStream { continuation in
+            eventsStreamContinuations[continuationID] = continuation
+            continuation.onTermination = { [weak self] _ in
+                self?.eventsStreamContinuations[continuationID] = nil
+            }
+        }
+    }
+
+    public func favoritesStream() -> AsyncStream<FavoriteList> {
+        let continuationID = UUID()
+        return AsyncStream { continuation in
+            favoriteStreamContinuations[continuationID] = continuation
+            continuation.onTermination = { [weak self] _ in
+                self?.favoriteStreamContinuations[continuationID] = nil
+            }
+        }
+    }
```

Comment:
While I like this pattern it's very repetitive. I'd like to come up with a way to reduce boilerplate for the same kind of async stream setup.

Reply:
Yeah, would be great to have in Dialpad as I imagine we will be writing it a lot in the future!